Sunset Legacy Sync (#1917)
casper-network/casper-node: node/CHANGELOG.md

* More node modules are now `pub(crate)`.
* Chain automatically creates a switch block immediately after genesis or an upgrade.

### Deprecated

> * Deprecate the `starting_state_root_hash` field from the REST and JSON-RPC status endpoints.

Reviewer: It's now _removed_ I suppose.

Author: As per discussion, this is apparently a breaking change to remove it so I've reverted.

Reviewer: Looks like it's still being removed (see `status_feed.rs`).

Author: So I'm looking here and it's still in: https://github.com/xcthulhu/casper-node/blob/1ef261249fd8a542cc5607882737665ce055c096/node/src/types/status_feed.rs#L172

Where do you see it missing?

Reviewer: Hmm, it was removed when I checked. If it's back, that's fine.

* More node modules are now `pub(crate)`.
* Chain automatically creates a switch block immediately after genesis or an upgrade.

### Deprecated

> * The `starting_state_root_hash` field from the REST status endpoint has been deprecated

Reviewer 2: ```suggestion
* Deprecate the `starting_state_root_hash` field from the REST and JSON-RPC status endpoints.
```

### Removed
* The unofficial support for nix-related derivations and support tooling has been removed.
* Experimental, nix-based kubernetes testing support has been removed.
* Removed dead code revealed by making modules `pub(crate)`.

> * Removed undocumented exit codes corresponding to `SIGINT`, `SIGTERM`, etc.

Reviewer 2: I'd prefer if this change was reverted, meaning this bullet point should also be removed.

If we decide to not revert those changes, then let's remove the word "undocumented" as they are currently documented.  Also, let's replace ", etc" with a complete list.

Author: I removed this because as far as I can tell the launcher is not using them. These are also non-standard... YAGNI?

Reviewer 2: The launcher uses `SIGTERM` to kill the node if it itself panics or receives one of the three termination signals.

> These are also non-standard

How so?  They're standard Unix signals.

If this change isn't really required by fast-sync, then it shouldn't be part of this PR.

Author: Okay, it looks like we are following a *nix convention here, and this exit code is getting logged in the launcher: https://github.com/casper-network/casper-node-launcher/blob/main/src/utils.rs#L117

I am non-plussed about how the launcher exit codes are different, but we can leave this in.

Author: Fixed in d9597e7b74b7776cb33b6cbaadb9afa8cae1d416

> * Removed legacy `LinearChainSync`

Reviewer 3: The user won't know what that means. Maybe:

```suggestion
* Removed legacy synchronization from genesis in favor of fast sync.
```

> * Removed legacy synchronization from genesis in favor of fast sync.

Reviewer 2: ```suggestion
* Remove dead code revealed by making modules `pub(crate)`.
* Remove undocumented exit codes corresponding to `SIGINT`, `SIGTERM`, etc.
* Remove legacy synchronization from genesis in favor of fast sync.
```

* Limit the maximum number of clients connected to the event stream server via the `[event_stream_server][max_concurrent_subscribers]` config option.
* Avoid emitting duplicate events in the event stream.
* Change `BlockIdentifier` params in the Open-RPC schema to be optional.

> * Asymmetric connections are now swept regularly again.

Reviewer: What happened here?

Author: These were fixed but not in 1.3.0 (it's broken there).

Looks like I should move this up to the `Unreleased` section.

Author: Fixed in d6c33b569
